### PR TITLE
Prevent future date in Customer DOB field

### PR DESCRIPTION
**Pull request by vishal-7037**
### Description (*)

Prevented future date in Customer DOB field at admin panel

### Fixed Issues (if relevant)

1. magento/magento2#22692: Add New Customer from admin panel - DOB field validation missing for future dates

### Manual testing scenarios (*)

1. Go to magento Admin panel
2. Go to Customer > All customers and Add new customer
3. Add required field values
4. Add Future date as DOB and save customer.

**After Fixed:** 

magento_datepicker

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by vishal-7037** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @vishal-7037. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @vishal-7037, here is your [new Magento instance](https://pr-22722.instances.magento-community.engineering).
Admin access: [https://pr-22722.instances.magento-community.engineering/admin](https://pr-22722.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by vishal-7037** [comment]
I've tested this changes On Vanilla Magento 2.3 instance  and seems working fine.

Vanilla_Customer_Dob

Thanks

File: app/code/Magento/Customer/view/base/ui_component/customer_form.xml [hunk 1]
```diff
@@ -260,13 +260,17 @@
...
+                        <item name="maxDate" xsi:type="string">new Date()</item>
+                    </item>
                 </item>
             </argument>
             <settings>
                 <validation>
                     <rule name="validate-date" xsi:type="boolean">true</rule>
                 </validation>
                 <dataType>text</dataType>
+
```
Comment (ronak2ram) [inline app/code/Magento/Customer/view/base/ui_component/customer_form.xml]: Hi @vishal-7037 
Please remove the extra added line.

**Comment by vishal-7037** [comment]
Hi @ronak2ram,

I've committed suggested changes Please review and let me know.

Thanks.

**Comment by ronak2ram** [comment]
Hi @vishal-7037 
Thanks for the update.

**Review by ronak2ram — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ronak2ram, thank you for the review.
[ENGCOM-5065](https://jira.corp.magento.com/browse/ENGCOM-5065) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Before:**

before

**After:**

after

**Comment by Nazar65** [comment]
Hi @vishal-7037 thank you for you pull request, this is wrong fix, because integration test fails, can you please refactor as in this example. thank you.
DeepinScreenshot_select-area_20190507164602

**Comment by vishal-7037** [comment]
Hi @Nazar65,

I've updated suggested changes and committed my changes. Please review it and let me know.

Thanks! :)

File: app/code/Magento/Customer/view/base/ui_component/customer_form.xml [hunk 2]
```diff
@@ -269,6 +269,15 @@
                 <dataType>text</dataType>
                 <visible>true</visible>
             </settings>
+            <formElements>
+                <date>
+                    <settings>
+                        <options>
+                            <option name="maxDate" xsi:type="string">new Date()</option>
```
Comment (VladimirZaets) [inline app/code/Magento/Customer/view/base/ui_component/customer_form.xml:276]: Hi, @vishal-7037. Thanks for collaboration. XML declaration shouldn't contain any code that must be executed.   If we really need it, you can do it in the javascript class related to this component.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi, @vishal-7037. Please make changes according to review.

**Comment by VladimirZaets** [comment]
@vishal-7037 , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**